cristin-person/src/main/java/no/unit/nva/cristin/person/update/UpdateCristinPersonHandler.java

```diff
@@ -44,16 +50,51 @@ protected Void processInput(String input, RequestInfo requestInfo, Context conte
         validateHasAccessRights(requestInfo);
 
         ObjectNode objectNode = readJsonFromInput(input);
-        PersonPatchValidator.validate(objectNode);
-        ObjectNode cristinJson = new CristinPersonPatchJsonCreator(objectNode).create().getOutput();
+        String personId = getValidPersonId(requestInfo);
 
+        if (clientCanUpdateAllFields(requestInfo)) {
+            PersonPatchValidator.validate(objectNode);
+            ObjectNode cristinJson = new CristinPersonPatchJsonCreator(objectNode).create().getOutput();
+            checkHasFields(cristinJson);
+
+            return apiClient.updatePersonInCristin(personId, cristinJson);
+        } else if (clientCanUpdateOwnData(requestInfo)) {
+            String personIdFromCognito = parseLastPartOfPersonCristinIdFromCognito(requestInfo).orElseThrow();
+            checkIdentifiersMatch(personId, personIdFromCognito);
+            PersonPatchValidator.validateOrcidIfPresent(objectNode);
+            ObjectNode cristinJson =
+                new CristinPersonPatchJsonCreator(objectNode).createWithAllowedUserModifiableData().getOutput();
+            checkHasFields(cristinJson);
+
+            return apiClient.updatePersonInCristin(personIdFromCognito, cristinJson);
+        } else {
+            throw new ForbiddenException();
+        }
+    }
+
+    private void checkHasFields(ObjectNode cristinJson) throws BadRequestException {
         if (cristinJson.isEmpty()) {
             throw new BadRequestException(ERROR_MESSAGE_NO_SUPPORTED_FIELDS_IN_PAYLOAD);
         }
+    }
 
-        String personId = getValidPersonId(requestInfo);
+    private void checkIdentifiersMatch(String personId, String personIdFromCognito) throws BadRequestException {
+        if (!Objects.equals(personId, personIdFromCognito)) {
```

Comment:
Maybe extract this as a method since it's difficu!t to see hidden exc!amation marks

Reply:
It is almost the only thing in this method. Should i extract it to another method? Do you have a suggestion on how to make the exclamation mark more visible?

Reply:
````
private boolean notEqual(String personId, String personIdFromCognito) {
    return !Objects.equals(personId, personIdFromCognito);
}
````